control_protocol.md

```diff
@@ -0,0 +1,359 @@
+`N1`, `N2`, etc. indicate Node Namespaces and `Co1`, `Co2` their corresponding Coordinators.
+
+Here the Message content is expressed in plain English, for the exact definition see {ref}`message-layer`.
+
+In the exchange of messages, only the messages over the wire are shown, the connection identity of the ROUTER socket is not shown.
+
+
+#### Communication with the Coordinator
+
+(sign-in)=
+##### Initial connection
+
+After connecting to a Coordinator (`Co1`), a Component (`CA`) shall send a SIGNIN message indicating its Component Name.
+The Coordinator shall respond with an ACKNOWLEDGE, giving the Namespace and other relevant information, or with an ERROR, if the Component Name is already taken.
+After that successful handshake, the Coordinator shall store the connection identity and corresponding Component Name in its local {ref}`directory`.
+It shall also publish to the other Coordinators in the network that this Component signed in, see {ref}`Coordinator coordination<coordinator-coordination>`.
+Similarly, the Component shall store the Namespace and use it from this moment to generate its full name.
+
+If a Component does send a message to someone without having signed in, the Coordinator shall refuse message handling and return an error.
+
+:::{mermaid}
+sequenceDiagram
+    Note over CA,N1: Name "CA" is still free
+    participant N1 as N1.COORDINATOR
+    CA ->> N1: V|COORDINATOR|CA|H|SIGNIN
+    Note right of N1: Connection identity "IA"
+    Note right of N1: Stores "CA" with identity "IA"
+    N1 ->> CA: V|N1.CA|N1.COORDINATOR|H|ACKNOWLEDGE: Namespace is "N1"
+    Note left of CA: Stores "N1" as Namespace
+    Note over CA,N1: Name "CA" is already used
+    CA ->> N1: V|COORDINATOR|CA|H|SIGNIN
+    N1 ->> CA: V|CA|N1.COORDINATOR|H|ERROR: Name "CA" is already used.
+    Note left of CA: May retry with another Name
+    Note over CA,N1: "CA" has not send SIGNIN
+    Note left of CA: Wants to send a message to CB
+    CA ->> N1: V|N1.CB|CA|H|Content
+    Note right of N1: Does not know CA
+    N1 ->> CA: V|CA|N1.COORDINATOR|H|ERROR:I do not know you
+    Note left of CA: Should send a SIGNIN message
+:::
+
+
+##### Heartbeat
+
+We use heartbeat to know, whether a communication partner is still online.
+
+Every message received counts as a heartbeat.
+
+A Component should and a Coordinator shall send a PING and wait some time before considering a connection dead.
+A Coordinator shall follow the {ref}`sign out routine<signing-out>` for a signed in Component considered dead.
+
+:::{note}
+TBD: Respond to every non empty message with an empty one?
+:::
+
+
+##### Signing out
+
+A Component should tell a Coordinator, when it stops working, with a SIGNOUT message.
+The Coordinator shall ACKNOWLEDGE the sign out and remove the Name from its directory.
+It shall also publish to the other Coordinators in the network that this Component signed out, see {ref}`Coordinator coordination<coordinator-coordination>`.
+
```

Comment:
```suggestion
It shall also notify the other Coordinators in the network that this Component signed out, see {ref}`Coordinator coordination<coordinator-coordination>`.

```
"publish to" sounds like zmq PUB, I'm not sure that that is the intention.

Reply:
It was my intention, when I wrote it, as that was the state of discussion.

Now "notify" is better.